aten/src/ATen/BatchedFallback.cpp

```diff
@@ -34,6 +34,172 @@ static bool areAnyArgumentsTensorList(const FunctionSchema& schema) {
       [] (const Argument& arg) { return arg.type()->isSubtypeOf(ListType::ofTensors()); });
 }
 
+// Returns if an operator is in-place. An operator is inplace if:
+// 1. its name ends in '_'
+// 2. The first argument is a Tensor and it is being written to
+// 3. The first argument is being returned
+// 4. No other arguments are aliased
+// Here is an example of an in-place operator:
+// add_(Tensor(a!) self, Tensor other, *, Scalar alpha=1) -> Tensor(a!)
+static bool isInplaceOp(const c10::FunctionSchema& schema) {
+  if (!schema.is_mutable() || schema.name().back() != '_') {
```

Comment:
The underscore test here is not sound; double underscore operations indicate inplaceness by being `__iblah__`. The current canonical definition of inplaceness is in BaseOperatorName in tools/codegen/model.py

Reply:
Thanks for the catch! There's also schemas like `__ior__` that do in-place (as declared by the `i`)... I'll take a look at tools/codegen/model.py and try to copy that

Reply:
Deleted the schema.name().back() != '_' check.